### ISSUE TITLE
SwiftLint > 0.59.1 throws ViolationSeverity error

### ISSUE THREAD

**Issue by KVSRoyal** [realm/SwiftLint#6433]
### New Issue Checklist

- [x] I've Updated SwiftLint to the latest version.
- [x] I've searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues).

### Bug Description

A clear and concise description of what the bug is. Ideally, provide a small (but compilable) example code snippet that can be used to reproduce the issue.

When I run swiftlint 0.63.0 (or any version above 0.59.1) via Bazel, e.g. 
```bash
bazel test //plugins/beacon/swiftui:PlayerUIBeaconPluginSwiftLint
```
I see this error:

```bash
external/swiftlint+/Source/SwiftLintCore/Models/SeverityConfiguration.swift:19:14: error: cannot assign to property: 'severity' is a get-only property
17 |     /// - parameter severity: The severity that should be used when emitting violations.
18 |     public init(_ severity: ViolationSeverity) {
19 |         self.severity = severity
   |              `- error: cannot assign to property: 'severity' is a get-only property
20 |     }
21 | 

external/swiftlint+/Source/SwiftLintCore/Models/SeverityConfiguration.swift:27:22: error: cannot assign to property: 'severity' is a get-only property
25 |         if let severityString: String = configString ?? configDict?[$severity.key] as? String {
26 |             if let severity = ViolationSeverity(rawValue: severityString.lowercased()) {
27 |                 self.severity = severity
   |                      `- error: cannot assign to property: 'severity' is a get-only property
28 |             } else {
29 |                 throw .invalidConfiguration(ruleID: Parent.identifier)

external/swiftlint+/Source/SwiftLintCore/Models/ViolationSeverity.swift:3:13: error: type 'ViolationSeverity' does not conform to protocol 'AcceptableByConfigurationElement'
 1 | /// The magnitude of a `StyleViolation`.
 2 | @AcceptableByConfigurationElement
 3 | public enum ViolationSeverity: String, Comparable, CaseIterable, Codable, Sendable, InlinableOptionType {
   |             `- error: type 'ViolationSeverity' does not conform to protocol 'AcceptableByConfigurationElement'
 4 |     /// Non-fatal. If using SwiftLint as an Xcode build phase, Xcode will mark the build as having succeeded.
 5 |     case warning
```

I am specifically seeing this with the player-ui repo, in [this PR](https://github.com/player-ui/player/pull/779/files#diff-6136fc12446089c3db7360e923203dd114b6a1466252e71667c6791c20fe6bdcR45) migrating to Xcode 26+. The SwiftLint call is defined in a helper [here](https://github.com/player-ui/rules_player/blob/ccabd1d093a57ae0f0aefe86e14accf5dfe5caab/ios/private/common_utils.bzl#L160-L186):

```starlark
    native.sh_test(
        name = name + "SwiftLint",
        srcs = [":" + name + "_Lint"],
        visibility = ["//visibility:public"],
    )

    # Runs the SwiftLint as part of the build, if lint fails with serious violations defer the results for the test
    native.genrule(
        name = name + "_Lint",
        tools = [
            "@SwiftLint//:swiftlint",
        ],
        srcs = [":" + name + "_Sources"] + ["//:.swiftlint.yml"],
        outs = ["output.sh"],
        executable = True,
        testonly = True,
        visibility = ["//visibility:public"],
        cmd = """
      echo `$(location @SwiftLint//:swiftlint) --config $(location //:.swiftlint.yml) $(SRCS) || true` > lint_results.txt
      LINT=$$(cat lint_results.txt)

      echo '#!/bin/bash' > $(location output.sh)
      echo "echo '$$LINT'" > $(location output.sh)

      LINESWITHERROR=$$(echo grep error lint_results.txt || true)
      echo "exit $$(($$LINESWITHERROR) | wc -l)" >> $(location output.sh)
  """,
    )
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure): 0.63.0
* Xcode version (run `xcodebuild -version` to be sure): 26.1 on MacOS 15.7.2
* Installation method used (Homebrew, CocoaPods, building from source, etc): Bazel
* Configuration file:

```yml
disabled_rules:
    - unused_optional_binding
    - unused_setter_value
type_name:
    excluded:
        - T
line_length:
    warning: 160
    ignores_comments: true
function_body_length: 50
identifier_name:
    excluded:
        - id
        - file_input
        - text_input
        - check_box
        - drop_down
        - radio_button
        - search_input
included:
    - ios
    - plugins
```

Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? If so, paste their
relative paths and respective contents.
Not using.

**Comment by KVSRoyal** [comment]
Here is a smaller reproduction: https://github.com/KVSRoyal/mini-demo-bazel-8/pull/1. Just need to run `bazel run //:xcodeproj` and `bazel run -c opt @SwiftLint//:swiftlint` to reproduce.

**Comment by SimplyDanny** [comment]
> Here is a smaller reproduction: [KVSRoyal/mini-demo-bazel-8#1](https://github.com/KVSRoyal/mini-demo-bazel-8/pull/1). Just need to run `bazel run //:xcodeproj` and `bazel run -c opt @SwiftLint//:swiftlint` to reproduce.

I checked out your branch and ran the two commands. The result looks good:

```bash
INFO: Found 1 target...
Target @@swiftlint+//:swiftlint up-to-date:
  bazel-bin/external/swiftlint+/swiftlint
INFO: Elapsed time: 257.396s, Critical Path: 252.01s
INFO: 200 processes: 63 internal, 86 darwin-sandbox, 51 worker.
INFO: Build completed successfully, 200 total actions
INFO: Running command line: bazel-bin/external/swiftlint+/swiftlint
Linting Swift files in current working directory
Linting 'BazelApp.swift' (1/1)
/Users/danny/Stuff/mini-demo-bazel-8/Sources/BazelApp.swift:32:1: warning: Trailing Newline Violation: Files should have a single trailing newline (trailing_newline)
Done linting! Found 1 violation, 0 serious in 1 file.
```

**Comment by KVSRoyal** [comment]
Interesting. That rules out the repo itself having a set up issue 🤔 . I wonder if it's related to my OS? I'm running MacOS 15 still. Will work on reproducing

**Comment by SimplyDanny** [comment]
I'm on macOS 26.2. So that's a difference.

**Comment by KVSRoyal** [comment]
Okay, so I got a friend to test out the mini demo for me too on their laptop, and here's what I learned, including your data:

| Works | MacOS | Xcode | Source |
| -- | -- | -- | -- |
| ✅  | 26.2 | 26.1? | SimplyDanny |
| ✅  | 15.6.1 | 16.4 | Friend |
| ❌ | 15.6.1 | 26.2 | Friend |
| ❌ | 15.7.2 | 26.1 | Me |
| ❌  | *15.6 | 26.2 | [Player PR](https://github.com/player-ui/player/pull/779/files#diff-6136fc12446089c3db7360e923203dd114b6a1466252e71667c6791c20fe6bdcR45), CircleCI |

My hypothesis is that it needs MacOS 26+ AND Xcode 26+ to work for some reason. Neither me nor my friend can update to MacOS 26 yet to check, unfortunately.

*Info added Jan 21

**Comment by KVSRoyal** [comment]
Fwiw, I'm now pretty sure upgrading to use MacOS 26 with Xcode 26+ fixes the issue.

**Comment by SimplyDanny** [comment]
> Fwiw, I'm now pretty sure upgrading to use MacOS 26 with Xcode 26+ fixes the issue.

Any idea why that is?
